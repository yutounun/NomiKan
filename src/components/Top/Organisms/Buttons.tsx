import React from "react";

import RocketLaunchOutlinedIcon from "@mui/icons-material/RocketLaunchOutlined";
import LoginOutlinedIcon from "@mui/icons-material/LoginOutlined";
import HowToRegOutlinedIcon from "@mui/icons-material/HowToRegOutlined";
import MyButton from "components/Commons/Atoms/MyButton";
import { Link } from "react-router-dom";
import { Stack } from "@mui/material";

function Buttons() {
  const buttons = [
    {
      to: "/total-amount",
      label: "はじめる",
      icon: <RocketLaunchOutlinedIcon />,
    },
    {
      // TODO: update path
      to: "/total-amount",
      label: "会員登録",
      icon: <HowToRegOutlinedIcon />,
    },
    {
      // TODO: update path
      to: "/total-amount",
      label: "ログイン",
      icon: <LoginOutlinedIcon />,
    },
  ];
  return (
    <Stack direction="column" gap={2} pt={3} justifyContent="center">
      {buttons.map((button) => (
        <Link key={button.label} to={button.to} style={{ textDecoration: "none" }}>
          <MyButton
            startIcon={button.icon}
            variant="outlined"
            value={button.label}
            color="primary"
            size="large"
            sx={{ borderRadius: "10px", width: "200px" }}
          />
        </Link>
      ))}
    </Stack>
  );
}

export default Buttons;
