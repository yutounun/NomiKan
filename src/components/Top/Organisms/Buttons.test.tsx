import { render, screen } from "@testing-library/react";
import React from "react";
import Buttons from "./Buttons";

describe("Buttons", () => {
  it("renders correctly", () => {
    render(<Buttons />);
    expect(screen.getAllByTestId("button")[0]).toBeInTheDocument();
    expect(screen.getAllByTestId("button")[1]).toBeInTheDocument();
    expect(screen.getAllByTestId("button")[2]).toBeInTheDocument();
    expect(screen.getByText("はじめる")).toBeInTheDocument();
    expect(screen.getByText("会員登録")).toBeInTheDocument();
    expect(screen.getByText("ログイン")).toBeInTheDocument();
  });
  it("takes a snapshot", () => {
    const { asFragment } = render(<Buttons />);
    expect(asFragment()).toMatchSnapshot();
  });
});
