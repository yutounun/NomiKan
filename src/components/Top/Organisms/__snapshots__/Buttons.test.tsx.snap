// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons takes a snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-1pimwu-MuiStack-root"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge css-9y1egq-MuiButtonBase-root-MuiButton-root"
      data-testid="button"
      style="width: 200px;"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-startIcon MuiButton-iconSizeLarge css-htszrh-MuiButton-startIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="RocketLaunchOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 15c-.83 0-1.58.34-2.12.88C2.7 17.06 2 22 2 22s4.94-.7 6.12-1.88c.54-.54.88-1.29.88-2.12 0-1.66-1.34-3-3-3zm.71 3.71c-.28.28-2.17.76-2.17.76s.47-1.88.76-2.17c.17-.19.42-.3.7-.3.55 0 1 .45 1 1 0 .28-.11.53-.29.71zm10.71-5.06c6.36-6.36 4.24-11.31 4.24-11.31S16.71.22 10.35 6.58l-2.49-.5c-.65-.13-1.33.08-1.81.55L2 10.69l5 2.14L11.17 17l2.14 5 4.05-4.05c.47-.47.68-1.15.55-1.81l-.49-2.49zM7.41 10.83l-1.91-.82 1.97-1.97 1.44.29c-.57.83-1.08 1.7-1.5 2.5zm6.58 7.67-.82-1.91c.8-.42 1.67-.93 2.49-1.5l.29 1.44-1.96 1.97zM16 12.24c-1.32 1.32-3.38 2.4-4.04 2.73l-2.93-2.93c.32-.65 1.4-2.71 2.73-4.04 4.68-4.68 8.23-3.99 8.23-3.99s.69 3.55-3.99 8.23zM15 11c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
          />
        </svg>
      </span>
      はじめる
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge css-9y1egq-MuiButtonBase-root-MuiButton-root"
      data-testid="button"
      style="width: 200px;"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-startIcon MuiButton-iconSizeLarge css-htszrh-MuiButton-startIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="HowToRegOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0-6c1.1 0 2 .9 2 2s-.9 2-2 2-2-.9-2-2 .9-2 2-2zM5 18c.2-.63 2.57-1.68 4.96-1.94l2.04-2c-.39-.04-.68-.06-1-.06-2.67 0-8 1.34-8 4v2h9l-2-2H5zm15.6-5.5-5.13 5.17-2.07-2.08L12 17l3.47 3.5L22 13.91z"
          />
        </svg>
      </span>
      会員登録
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge css-9y1egq-MuiButtonBase-root-MuiButton-root"
      data-testid="button"
      style="width: 200px;"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-startIcon MuiButton-iconSizeLarge css-htszrh-MuiButton-startIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="LoginOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 7 9.6 8.4l2.6 2.6H2v2h10.2l-2.6 2.6L11 17l5-5-5-5zm9 12h-8v2h8c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-8v2h8v14z"
          />
        </svg>
      </span>
      ログイン
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
